app/bundles/CampaignBundle/Controller/Api/CampaignApiController.php

```diff
@@ -20,23 +27,37 @@ class CampaignApiController extends CommonApiController
 {
     use LeadAccessTrait;
 
-    /**
-     * @var MembershipManager
-     */
-    private $membershipManager;
+    private MembershipManager $membershipManager;
 
     /**
      * @var CampaignModel|null
      */
     protected $model = null;
 
+    private RequestStack $requestStack;
+
+    public function __construct(
+        CorePermissions $security,
+        Translator $translator,
+        EntityResultHelper $entityResultHelper,
+        RouterInterface $router,
+        FormFactoryInterface $formFactory,
+        AppVersion $appVersion,
+        RequestStack $requestStack,
+        MembershipManager $membershipManager,
+    ) {
+        $this->requestStack      = $requestStack;
+        $this->membershipManager = $membershipManager;
+
+        parent::__construct($security, $translator, $entityResultHelper, $router, $formFactory, $appVersion, $requestStack);
+    }
+
     public function initialize(ControllerEvent $event)
     {
         $campaignModel = $this->getModel('campaign');
         \assert($campaignModel instanceof CampaignModel);
```

Comment:
The ModelFactory is also getting the services directly from the container. We might have to refactor that next. But not as part of this PR. Just FYI...